### ISSUE TITLE
Some torrents paused and can't resumed if downloaded from magnet 

### ISSUE THREAD

**Issue by proninyaroslav** [proninyaroslav/libretorrent#93]

**Comment by proninyaroslav** [comment]
The developer of `jlibtorrent` is notified of this problem and I hope that soon it will be solved.

**Comment by proninyaroslav** [comment]
Workaround is deleting trackers from magnet link (e.g. `&tr=tracker_url`) or using torrent file.

**Comment by rak5381** [comment]
You asked for original magnet in #90 . First was posted by OP there. ("Magnet A')

magnet:?xt=urn:btih:59066769B9AD42DA2E508611C33D7C4480B3857B&dn=ubuntu-17.04-desktop-amd64.iso&tr=udp%3A%2F%2Ftracker.openbittorrent.com%3A80&tr=udp%3A%2F%2Ftracker.publicbt.com%3A80&tr=udp%3A%2F%2Ftracker.istole.it%3A6969&tr=udp%3A%2F%2Fopen.demonii.com%3A1337

The second ("Magnet B"), an updated magnet from https://thepiratebay.org/torrent/17809088/ubuntu-17.04-desktop-amd64.iso also doesn't work.

magnet:?xt=urn:btih:59066769b9ad42da2e508611c33d7c4480b3857b&dn=ubuntu-17.04-desktop-amd64.iso&tr=udp%3A%2F%2Ftracker.leechers-paradise.org%3A6969&tr=udp%3A%2F%2Fzer0day.ch%3A1337&tr=udp%3A%2F%2Fopen.demonii.com%3A1337&tr=udp%3A%2F%2Ftracker.coppersurfer.tk%3A6969&tr=udp%3A%2F%2Fexodus.desync.com%3A6969

**However**, any magnet containing a combination of **one**, **two**, or **three** of the trackers from Magnets A and/or B *does* download correctly. But any magnet combining four or more doesn't. (With the exception listed below).

Another source for the same download file ('Magnet C") uses **six** trackers and works fine:

magnet:?xt=urn:btih:59066769b9ad42da2e508611c33d7c4480b3857b&dn=ubuntu%2017%2004%20desktop%20amd64%20iso&tr=http://mgtracker.org:2710/announce&tr=udp://tracker.coppersurfer.tk:6969/announce&tr=udp://tracker.leechers-paradise.org:6969/announce&tr=udp://tracker.mg64.net:6881/announce&tr=udp://tracker.zer0day.to:1337/announce&tr=udp://tracker2.itzmx.com:6961/announce


Though the six items all include the URL ending "/announce", the magnet works just as well without them. And like Magnets A and B, one, two, or three of the URLs work properly. But unlike Magnets A and B, four, five,an six URLs from Magnet C work fine for downloading.


Noteworthy, adding the ending "/announce" to *any* of the three following URLs from Magnets A, B, and C (and of those trackers listed here it is only these three) allows any number of trackers Tobe present in a magnet URI and still function.

tracker.istole.it:6969/announce
tracker.publicbt.com:80/announce
tracker.openbittorrent.com:80/announce

-----

In summary, at most 3 trackers can be listed in a magnet file, unless it's in the presence of a special tracker like the three listed above or is one of those like in Magnet C.

What sets them apart I do not know.


Thank you

**Comment by rak5381** [comment]
The workaround of adding tracker.istole.it:6969/announce works for Rutracker in the meantime. Try it and see:

magnet:?xt=urn:btih:6EA43E8EF81170ECBACA6F82D4A0FF6480D56026&tr=http%3A%2F%2Fbt.t-ru.org%2Fann%3Fmagnet&tr=udp%3A%2F%2Ftracker.istole.it%3A6969/announce&dn=%D0%98%D1%81%D1%87%D0%B5%D0%B7%D0%BD%D0%BE%D0%B2%D0%B5%D0%BD%D0%B8%D0%B5%20%D0%A5%D0%B0%D1%80%D1%83%D1%85%D0%B8%20%D0%A1%D1%83%D0%B4%D0%B7%D1%83%D0%BC%D0%B8%D0%B8%20%2F%20Suzumiya%20Haruhi%20no%20Shoushitsu%20%2F%20The%20Disappearance%20of%20Haruhi%20Suzumiya%20%5BMovie%5D%20%5B%D0%91%D0%B5%D0%B7%20%D1%85%D0%B0%D1%80%D0%B4%D1%81%D0%B0%D0%B1%D0%B0%5D%20%5BRUS(int)%2C%20JAP%20%2B%20SUB%5D%20%5B2010

**Comment by rak5381** [comment]
More observations:

I open the problem magnets above: "Magnet A" or "Magnet B" or the Rutorrent magnet, wait for the metadata to download, and then it pauses as in the bug.

But If I go to the "Files" tab in the torrent, and set all files priority to "Low (ignore file)", then the download is marked completed and pauses.

If I unpause, it says "seeding" instead of just "paused", and it doesn't pause itself.

If I then go back to "files", change the priority to "normal" or "high" and start the torrent, it says "downloading" and starts to find and try to connect to peers. It still won't download from them, but it keeps trying to. Trying to "Force announce" did nothing.

If I then delete some or all the trackers from the "Tracker" tab, it still finds peers but doesn't download. If I then turn off DHT, LSD, and uTP, it still finds peers but doesn't actual download.

If I "Force recheck", it still gets stuck checking. Unless I pause, set files to "low priority", unpause, and return to "normal" priority. Then it goes back to trying to download.

**Comment by rak5381** [comment]
I found an even easier solution: Just restart the app. Works for all the above magnets.

**Comment by proninyaroslav** [comment]
@rak5381 
Yes, because it recreates `TorrentHandle`.

**Comment by lyn1337** [comment]
@rak5381 ty!

**Comment by proninyaroslav** [comment]
Fixed.
